pyspeckit/spectrum/plotters.py

```diff
@@ -341,19 +365,19 @@ def reset_limits(self,xmin=None, xmax=None, ymin=None, ymax=None,
             if ymax is not None: self.ymax = ymax
             elif self.ymax is None:
                 if hasattr(self.Spectrum.data, 'mask'):
-                    ymaxval = ((self.Spectrum.data[xpixmin:xpixmax]).max()-self.ymin)
+                    ymaxval = ((self.Spectrum.data[xpixmin:xpixmax]).max()-self.ymin.value)
                 else:
-                    ymaxval = (np.nanmax(self.Spectrum.data[xpixmin:xpixmax])-self.ymin)
+                    ymaxval = (np.nanmax(self.Spectrum.data[xpixmin:xpixmax])-self.ymin.value)
                 if ymaxval > 0:
-                    self.ymax = float(ymaxval) * float(ypeakscale) + self.ymin
+                    self.ymax = float(ymaxval) * float(ypeakscale) + self.ymin.value
                 else:
-                    self.ymax = float(ymaxval) / float(ypeakscale) + self.ymin
+                    self.ymax = float(ymaxval) / float(ypeakscale) + self.ymin.value
 
-            self.ymin += self.offset
-            self.ymax += self.offset
+            self.ymin += u.Quantity(self.offset, self.ymin.unit)
+            self.ymax += u.Quantity(self.offset, u.Unit(self.ymax.unit))
```

Comment:
These lines should look more similar; I think 376 is correct and 377 is unnecessary.